src/newusers.c

```diff
@@ -1130,16 +1130,23 @@ int main (int argc, char **argv)
 		 * First check if we have to create or update an user
 		 */
 		pw = pw_locate (fields[0]);
+		if (NULL == pw) {
+			fprintf (stderr,
+			         _("%s: line %d: user '%s' does not exist in %s\n"),
+			         Prog, line, fields[0], pw_dbname ());
+			errors++;
+			continue;
+		}
+		newpw = *pw;
+
 		/* local, no need for xgetpwnam */
-		if (   (NULL == pw)
-		    && (getpwnam (fields[0]) != NULL)) {
+		if (getpwnam (fields[0]) != NULL) {
```

Comment:
I don't think this set is right.  In the original code, these calls are only meant to be done if pw == NULL, that is, the user did not yet exist and needed to be added.  But after your patch these would always be done.

Reply:
You are completely right. I misunderstood what the code was doing.